src/components/3d_editor/advanced_geometry/python_transformation/PythonTransformation.jsx

```diff
@@ -0,0 +1,145 @@
+import Dialog from "@exabyte-io/cove.js/dist/mui/components/dialog/Dialog";
+import PyodideLoader from "@exabyte-io/cove.js/dist/other/pyodide";
+import theme from "@exabyte-io/cove.js/dist/theme";
+import ThemeProvider from "@exabyte-io/cove.js/dist/theme/provider";
+import Paper from "@mui/material/Paper";
+import Stack from "@mui/material/Stack";
+import PropTypes from "prop-types";
+import React from "react";
+import NPMsAlert from "react-s-alert";
+
+import MaterialsSelector from "./MaterialsSelector";
+import PythonCodeDisplay from "./PythonCodeDisplay";
+import PythonCodeExecution from "./PythonCodeExecution";
+import TransformationSelector from "./TransformationSelector";
+
+class PythonTransformation extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            materials: props.materials,
+            selectedMaterials: [props.materials[0]],
+            isLoading: true,
+            isRunning: false,
+            pyodide: null,
+            pythonCode: "",
+            pythonOutput: "",
+        };
+        this.handleRun = this.handleRun.bind(this);
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (prevState.pythonCode === "") {
```

Comment:
What's the logic here? What if I delete all the code, this will trigger the load of pyodide